route.go

```diff
@@ -72,7 +72,11 @@ func (r *Route) Match(req *http.Request, match *RouteMatch) bool {
 		return false
 	}
 
-	match.MatchErr = nil
+	if match.MatchErr == ErrMethodMismatch {
+		// We found a non-mismatching route, clear MatchErr
```

Comment:
non-mismatching is a bit of a double negative :) how about just "matching" so it's less confusing?

Reply:
Hm, I used that wording to make clear that the ErrMethod*Mismatch* found on the previous iteration no longer applies. But I guess I can reword it a bit.

I'll do it tomorrow tho, its 2AM here and I'm quite asleep.